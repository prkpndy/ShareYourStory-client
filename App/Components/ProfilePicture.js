import React from 'react';
import {Image} from 'react-native';

const ProfilePicture = props => {
    if (props.isProfilePictureDownloaded) {
        return (
            <Image
                style={props.dimensions}
                source={{uri: `file://${props.profilePictureDetails.path}`}}
            />
        );
    }
    return (
        <Image
            style={props.dimensions}
            source={require('../res/images/noProfilePicture.png')}
        />
    );
};

export default ProfilePicture;
