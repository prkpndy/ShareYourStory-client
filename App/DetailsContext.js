import React from 'react';

const DetailsContext = React.createContext();

export default DetailsContext;
