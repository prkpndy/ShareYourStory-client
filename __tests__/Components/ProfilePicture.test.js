/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProfilePicture from '../../App/Components/ProfilePicture';
import DetailsContext from '../../App/DetailsContext';

const profilePictureDetails = {
    path: 'res/images/noProfilePicture.png',
    type: 'image/png',
    name: 'noProfilePicture.png',
};

it('renders correctly', () => {
    const tree = renderer
        .create(
            <DetailsContext.Provider
                value={{
                    isProfilePictureDownloaded: false,
                    profilePictureDetails,
                }}>
                <ProfilePicture
                    dimensions={{height: 200, width: 200}}
                    borderColor={'bright'}
                />
            </DetailsContext.Provider>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
