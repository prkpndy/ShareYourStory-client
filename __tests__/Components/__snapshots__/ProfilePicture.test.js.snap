// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../App/res/images/noProfilePicture.png",
    }
  }
  style={
    Object {
      "borderColor": "#fcf400",
      "borderWidth": 4,
      "height": 200,
      "padding": 5,
      "width": 200,
    }
  }
/>
`;
